codebase/superdarn/src.lib/tk/fitacf_v3.0/src/preprocessing.c

```diff
@@ -823,7 +823,7 @@ double ACF_cutoff_pwr(FITPRMS *fit_prms){
     ni = (ni > 0) ? ni :  1;
     cpc = cutoff_power_correction(fit_prms);
     min_pwr = min_pwr/ni * cpc;
-    if (min_pwr < 1.0) min_pwr = fit_prms->noise;
+    /*if (min_pwr < 1.0) min_pwr = fit_prms->noise;*/
```

Comment:
Do you want to add a brief comment here to explain why this line was commented out?

Reply:
Thanks a lot for leading me through the correct procedures, @ecbland ! 
:-) 

On the sanity check, I am not sure what should be the output of this procedure if the background noise value is zero for some reason (probably, only possible for older data).

On a more general note, the original replacement of "sky" noise < 1  with the "search" noise seems to be a bit strange. The latter is estimated through a completely different procedure involving FFT which is performed over a short time interval and, I suspect, at a higher sampling rate. I have had a look at the code used by ROS, and there are virtually no comments throughout the text and some interesting modification remarks like this:

_Revision 1.7  2004/05/14 16:27:31  barnes
Divided noise value by three to correct for signal abnormalities._

I assume that some sort of _ad hoc_ scaling has been applied to the "search" noise level in order to match that of the "sky" noise. From what I have seen, this scaling is not working for at least some datasets. On the other hand, this replacement might still be required for processing old datasets... Looks like we might need to open another issue to dig deeper into this.

Reply:
Further on the noise value replacemet, I feel that we may need to re-think the original way of determining the "sky" noise that arbitrarily utilises only one third of the available range gates:

 /*  look for the lowest 10 values of lag0 power and average to
            get the noise level.  Ignore values that are exactly 0.  If
            you can't find 10 usable values within the first 1/3 of the
            sorted power list, then just use whatever you got in that
            first 1/3.  If you didn't get any usable values, then use
            the NOISE parameter */

    while ((ni < 10) && (i < fit_prms->nrang/3)) {
        if (pwr_levels[i]) ++ni;
        min_pwr += pwr_levels[i++];
    }

Reply:
> I assume that some sort of ad hoc scaling has been applied to the "search" noise level in order to match that of the "sky" noise.

Oh dear :rofl: :rofl:

> we may need to re-think the original way of determining the "sky" noise that arbitrarily utilises only one third of the available range gates

I've also been wondering about this. It sounds like another ad-hoc legacy thing, perhaps from the days when not all pwr0 values were stored??  We need to be careful that it doesn't break anything for the really old data.

Reply:
Yep, there was another layer of pre-selection by the ROS when the data with lag 0 power below a certain A/D converison level were not stored at all. I tried to find how the rejection threshold was calculated but did not succeed as there were no documentation on this. 

Zero values in raw data have been discovered by Keith when he was testing FITACF3 and obtaining infite spectrl width for some older data.

I also recall that ANge,ine might have had a closer look at this issue as well but don't remeber the details....